### ISSUE TITLE
Error on fresh install

### ISSUE THREAD

**Issue by deejayxd07** [parmsang/gatsby-starter-ecommerce#1]
Hi, I've found this project very nice and interesting.

Just curious if this was an expected behavior on fresh install.

`TypeError:` Cannot read property 'link' of null`

![image](https://user-images.githubusercontent.com/26223130/41498627-392ed1c0-71a5-11e8-98da-9495b21f748f.png)

Cheers!

**Comment by deejayxd07** [comment]
Hi, I've created a PR for the fix.

[https://github.com/parmsang/gatsby-starter-ecommerce/pull/2/commits/abcfe63203e5b5a69b34d091955a5bacdcf14b54](url)

**Comment by deejayxd07** [comment]
Hi, I've created a PR for the fix.

It's on abcfe63

**Comment by parmsang** [comment]
Hi, Thanks for the positive feedback! 

That isn't the expected behavior. The reason for it is, rather unexpectedly, the source of the data (i.e. moltin) updated the response data and the number of products being returned. The products  increased and included some without images, hence the error. I've updated the code to address this issue before I was aware of this PR.  

I've also used gatsby-image for the home images in the latest update. Things should run more smoothly!
